website/content/docs/discovery/dns.mdx

```diff
@@ -285,6 +285,50 @@ caching of DNS results. However, there are many situations in which caching is
 desirable for performance and scalability. This is discussed more in the tutorial
 for [DNS caching](https://learn.hashicorp.com/tutorials/consul/dns-caching).
 
+## DNS with ACLs
+
+After enabling [Access Control Lists (ACLs)](/docs/security/acl/acl-system),
+DNS requests will fail with error `NXDOMAIN` unless ACL tokens have been
+configured with the necessary rules.
+
+When a DNS request is made to a Consul agent, the agent will use a previously
+configured token with the request (since there is no means for the initial DNS
+request to specify a token):
+
+- The agent's [`default` token](/docs/agent/options#acl_tokens_default)
+  token, if configured.
+- Otherwise, the built-in [`anonymous` token](/docs/security/acl/acl-system#builtin-tokens).
+  Because the anonymous token is used when any request is made to Consul without
+  explicitly specifying a token, production deployments should not apply policies
+  needed for DNS to this token.
+
+Consul will either authorize or revoke the request, depending on the token's
+policy rules:
+
+- [Node lookups](#node-lookups) (`*.node.consul`) require
+  [`node:read`](/docs/security/acl/acl-rules#node-rules) permission for the
+  target node.
+- [Service lookups](#service-lookups) (`*.service.consul`, `*.query.consul`,
+  `*.connect.consul`, `*.ingress.consul`) require:
+  - [`service:read`](/docs/security/acl/acl-rules#service-rules) permission for
+    the target service, and
+  - [`node:read`](/docs/security/acl/acl-rules#node-rules) permission for nodes
+    on which the target service instances are running.
+- [Prepared query lookups](#prepared-query-lookups) (`*.query.consul`) are a
+  special type of service lookup that require:
+  - [`query:read`](/docs/security/acl/acl-rules#prepared-query-rules)
+    permission for the target prepared query, and
+  - the same permissions as needed for a service lookup of the query's
+    [target service](/api/query#service-1)
+    ([`service:read`](/docs/security/acl/acl-rules#service-rules),
+    [`node:read`](/docs/security/acl/acl-rules#node-rules)).
```

Comment:
This would probably be easier to read as a table.

Reply:
I'll try this out and see what it looks like. Then we can decide between the options.

Reply:
It's not clear how this information completes the idea in the lead-in statement ("Consul will authorize the request depending on the token's policy rules"). I would expect more specific guidance on how to configure the policy rules to enable authorization. It takes a lot of mental work to put align the lookup options with the goal of request authorization. An example configuration based on some kind of real-world scenario would help.

Reply:
@trujillo-adam : I agree, I'll see if I can come up with a real-world scenario + configuration to help. Regardless, I think these details are needed (either before or after the example).

@blake : I tested out reformatting this information as a table. What do you think?

![image](https://user-images.githubusercontent.com/85913323/132796898-a0ae601b-007e-458d-90f3-d3d2078fcf2a.png)

Reply:
@jkirschner-hashicorp what are the questions marks in the title for ? 
I do like the table. We can probably do away with the parentheses around the code blocks.

Reply:
I originally didn't have the question marks, so the title was just `Requires service:read`. Having an answer "yes" in a cell without an explicit question being asked felt weird, so I tried out adding a question mark to the title. The question mark look funny though, and it seems clear enough in context without it. I'll remove it.

It feels weird that `*.query.consul` is listed in two rows. It's a special case of service lookup. **Q:** What do we think about changing "Service Lookups" to "Standard Service Lookups" and "Prepared Query Lookups" to "Prepared Query Service Lookups"?

Reply:
In order to be consistent with how ACL policy requirements are expressed throughout the rest of the site, it might be better to collapse the permission requirements into a single "ACL Required" column. An additional "Description" or similar column can be added to explain why the policies are needed.

For example

| Lookup type | ACLs required | Description |
| --- | --- | --- |
| Node lookups (`*.node.consul`) | `node:read` | Allow resolving the IP address for the target node |
| Service lookups (`*.service.consul`) | `service:read` <br> `node:read` | Allow resolving the IP addresses for the target service and node on which the service instances are running. |

Reply:
I just pushed a commit where I tried this out (see below). I think it's easier to read, though I'm not sure the connection between the ACL rules and the behavior is as clear as the previous table option.

![image](https://user-images.githubusercontent.com/85913323/132889504-be47ad80-0c50-4dc9-a9ca-10cf8092d930.png)

Perhaps that can be overcome with an example like @trujillo-adam was suggesting?

Reply:
IMO the updated table has a reduced cognitive load compared to the previous table. 👍  for adding a code example to show different policy combinations.

Reply:
I think I see 4 columns:

| Lookup           | Type    | Description                                                                                                    | Required ACLs          |
|------------------|---------|----------------------------------------------------------------------------------------------------------------|------------------------|
| *.node.consul    | Node    | Allows IP addresses to be resolved for the target node                                                         | node:read              |
| *.service.consul | Service | Allows IP addresses to be resolved for the target service and node on which the service instances are running. | service:read node:read |
| *.query.consul   | Service | Allows IP addresses to be resolved for the target service and node on which the service instances are running. | service:read node:read |

I also think that each lookup should get its own row, even if there is repetition within the table. 
I didn't understand what the information was trying to express until I saw @Blake 's suggestion

Reply:
@trujillo-adam : thanks for the suggestion - I think that version is even more readable.

One part I'm not sure I agree with is "each lookup should get its own row, even if there is repetition within the table". When there's lots of repetition, it's hard to spot differences. If we have a row for `*.service.consul`, `*.connect.consul`, `*.ingress.consul`, then `*.query.consul`, it will require a careful read to see that the first three are identical, but the last one has subtle differences.

Unfortunately, with the table component, there's no way to make a cell span multiple rows or columns.... which would allow us to have one shared cell across the 3 identical rows. Given that this isn't an option, I think keeping service, connect, and ingress style service lookups in the same row makes sense.

I'll try out what I'm thinking and then post the image.

Reply:
Here's the latest based on all your feedback

![image](https://user-images.githubusercontent.com/85913323/132915330-7b3fec8e-16bf-4ccf-908a-d406bbf32d45.png)

Reply:
That is a good looking table!